.github/workflows/gh001-build-push-dev.yaml

```diff
@@ -0,0 +1,99 @@
+name: build-push-dev
+
+on:
+  workflow_call:
+    inputs:
+      oidc_role:
+        required: true
```

Comment:
What do you think of separating the build and push from deployment?

I'm thinking that that means that release can re-use the same workflow unless the tests and validations are going to be different.

Reply:
@hugoalmeida264 by having build, push and deploy to dev together, we deploy strictly from the `rc` tag that triggered the workflow. i.e we tag the images using `github.ref_name `(which is the tag that triggered the workflow)

For the release workflow, we use the tag output from the release job to tag docker images. So it may not be possible to reuse docker build and push jobs in these 2 scenarios.

On the other hand, the workflow that deploys depends on the last release that was made. We query the tag associated with the most recent release, then used the image that has that same tag.

Please let me know what you think.